pymc3/tests/test_sg.py

```diff
@@ -0,0 +1,26 @@
+import numpy as np
+
+from pymc3.blocking import DictToArrayBijection
+from . import models
+from pymc3.step_methods.hmc.base_hmc import BaseHMC
+from pymc3.step_methods.hmc import SGFS
+import pymc3
+from pymc3.theanof import floatX
+from .checks import close_to
+from theano import tensor
+
+
+def example_model():
+    pm = pymc3
+    cT = tensor.vector()
+    batch_size = 10
+    total_size = 100
+    cT.tag.test_value = floatX(np.ones(batch_size))
+    with pm.Model() as model:
+        a = pm.Normal('a', mu=0, sd=1.0)
+        b = pm.Normal('b', mu=1.0, sd=1.0)
+        c = pm.Normal('c', mu=a + b, sd=1.0, observed=cT)
```

Comment:
I considered the following interface very flexible. Let user feel free of what mini batch interface to use. There are three options.

0. Set `total_size=N` for observed and local vars
1. Set minibatch generator via `pm.generator`. It is very easy, requires minimum code. But sometimes corner case happen. See https://github.com/Theano/Theano/issues/5809
2. Use shared variable in advance for data storing and define a callback that changes storage. Requires `callbacks` kwarg. It is also useful for getting custom statistics from the process, see [this notebook](https://github.com/ferrine/pymc3/blob/a62b4fa98e75c392e634546d29cd5f2266007c46/docs/source/notebooks/simple_stochastic_optimization.ipynb) for example.
3. use `more_replacements` kwarg. Usually it replaces input with `pm.generator` to avoid https://github.com/Theano/Theano/issues/5809

I hope in future we'll have consistent interface using this paradigm for scalability